### PR TITLE
🌱 clusterctl upgrade test

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
Adding clusterctl upgrade test pipeline

### CODE REVIEW

**Comment by adilGhaffarDev** [comment]
cc: @Sunnatillo @mboukhalfa  can you check this?

File: jenkins/jobs/clusterctl_upgrade.pipeline [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    agent none
+    environment {
+        METAL3_CI_USER = 'metal3ci'
+        REPO_ORG = "${PROJECT_REPO_ORG}"
+        REPO_NAME = "${PROJECT_REPO_NAME}"
+        UPDATED_REPO = "${ghprbAuthorRepoGitUrl}"
+        REPO_BRANCH = "${ghprbTargetBranch}"
+        UPDATED_BRANCH = "${ghprbActualCommit}"
+        BUILD_TAG = "${env.BUILD_TAG}"
+        PR_ID = "${ghprbPullId}"
```
Comment (mboukhalfa) [inline jenkins/jobs/clusterctl_upgrade.pipeline:41]: where this can be used

File: jenkins/jobs/clusterctl_upgrade.pipeline [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+ci_git_credential_id = "metal3-jenkins-github-token"
+
+// 3 hours
+int TIMEOUT = 10800
+
+KEEP_TEST_ENV = (env.KEEP_TEST_ENV)
+
+script {
+    if ("${PROJECT_REPO_ORG}" == 'metal3-io' && "${PROJECT_REPO_NAME}" == 'project-infra') {
```
Comment (mboukhalfa) [inline jenkins/jobs/clusterctl_upgrade.pipeline:12]: Do we need to run clusterctl upgrade on project infra

File: jenkins/jobs/clusterctl_upgrade.pipeline (hunk 1 above)
Comment (adilGhaffarDev) [reply jenkins/jobs/clusterctl_upgrade.pipeline:41]: REPO_BRANCH is the branch where we are running the test(main,1.6 etc) and UPDATED_BRANCH is the commit on which we want to run test. BUILD_TAG is used to name the log folders. PR_ID I am not sure why we need it, I haven't been find where this is being used. I will remove it and see if anything breaks.

File: jenkins/jobs/capm3-e2e-tests.pipeline [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        REPO_BRANCH = "${ghprbTargetBranch}"
+        UPDATED_BRANCH = "${ghprbActualCommit}"
+        BUILD_TAG = "${env.BUILD_TAG}"
+        IMAGE_OS = "${IMAGE_OS}"
+        CAPM3RELEASEBRANCH = "${capm3_release_branch}"
+        BMORELEASEBRANCH = "${bmo_release_branch}"
+        TARGET_NODE_MEMORY = "${TARGET_NODE_MEMORY}"
+        NUM_NODES = 4
+        TESTS_FOR = "${TESTS_FOR}"
+        KEEP_TEST_ENV = false
```
Comment (Sunnatillo) [inline jenkins/jobs/capm3-e2e-tests.pipeline:41]: remove it from here as well

File: jenkins/jobs/capm3-e2e-tests.pipeline [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import java.text.SimpleDateFormat
+import java.security.SecureRandom
+
+ci_git_credential_id = "metal3-jenkins-github-token"
+
+// 3 hours
+int TIMEOUT = 10800
+
+KEEP_TEST_ENV = (env.KEEP_TEST_ENV)
```
Comment (Sunnatillo) [inline jenkins/jobs/capm3-e2e-tests.pipeline:9]: Can you remove KEEP_TEST_ENV var from here? it is not used in dynamic worker case.

File: jenkins/jobs/capm3-e2e-tests.pipeline [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        REPO_NAME = "${PROJECT_REPO_NAME}"
+        UPDATED_REPO = "${ghprbAuthorRepoGitUrl}"
+        REPO_BRANCH = "${ghprbTargetBranch}"
+        UPDATED_BRANCH = "${ghprbActualCommit}"
+        BUILD_TAG = "${env.BUILD_TAG}"
+        IMAGE_OS = "${IMAGE_OS}"
+        CAPM3RELEASEBRANCH = "${capm3_release_branch}"
+        BMORELEASEBRANCH = "${bmo_release_branch}"
+        TARGET_NODE_MEMORY = "${TARGET_NODE_MEMORY}"
+        NUM_NODES = 4
```
Comment (Sunnatillo) [inline jenkins/jobs/capm3-e2e-tests.pipeline:39]: I believe it NUM_NODES should be 5 for clusterctl_upgrade tests.

Comment (adilGhaffarDev) [reply jenkins/jobs/capm3-e2e-tests.pipeline:39]: yes, but we change it again on CAPM3 side, I will just remove it from here

Comment (Sunnatillo) [reply jenkins/jobs/capm3-e2e-tests.pipeline]: sure. Since we are planning to use it for all e2e tests it makes sense.

**Comment by Sunnatillo** [comment]
let's hold untill jjb gets merged, then we will be able to tests these changes here. 
/hold

**Comment by adilGhaffarDev** [comment]
> let's hold until jjb PR gets merged, then we will be able to tests these changes here.

we cannot run clusterctl upgrade job on this repo.

**Comment by Sunnatillo** [comment]
> > let's hold until jjb PR gets merged, then we will be able to tests these changes here.
> 
> we cannot run clusterctl upgrade job on this repo.

why not?

**Comment by adilGhaffarDev** [comment]
@Sunnatillo added project infra back

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Review by mboukhalfa — APPROVED** [review]
lgtm

**Pull request merged**